index.js

```diff
@@ -1,26 +1,32 @@
 import axios from "axios";
 import dotenv from "dotenv";
+import fs from "fs";
 dotenv.config();
 
 const riotApi = process.env.API_URL;
 const baseUrl = "https://eu.api.riotgames.com/val/content/v1/contents"
 
+let datas;
 const getData = async () => {
     const url = `${baseUrl}?api_key=${riotApi}`;
-
     try {
         const response = await axios.get(url)
             .then((res) => {
-                const data = JSON.stringify(res.data);
-                console.log(data);
+
+                if (res.status === 429) {               // if api exceeds the limit, read the data from the file
+                    const readData = JSON.parse(fs.readFileSync("output.json"));
+                    datas = readData;
+                    return
+                }
+                else {
+                    datas = (res.data);
+                    return
+                }
             })
     } catch (error) {
         console.error
-
     }
-
-
-
 }
+await getData();
```

Comment:
Please check on this as await is only used inside async functions, so i don't think it is required, but i could be wrong

Reply:
we can use await to call async function. But here it doesn't makes sense. 
Thanks for the suggestion.
await removed.

Reply:
I am having error if not using await. 
As the line below calling function is dependent on the data from the function. The data is undefined. And using await helped solve this issue.
it is behaving asynchronous the line below is executing before the function is fully executed.

Reply:
```
const data = await getData();
```